js/wp-seo-replacevar-plugin-302.js

```diff
@@ -121,23 +189,167 @@
+			}
+		}.bind( this ) );
+
+		return data;
+	};
+
+	/**
+	 * Returns the string to replace the taxonomy var.
+	 *
+	 * @param {String} taxonomy
+	 * @returns {String}
+	 */
+	YoastReplaceVarPlugin.prototype.getTaxonomyReplaceVar = function( taxonomy ) {
+		if ( this.currentTaxonomies[ taxonomy ] !== undefined ) {
+			return jQuery.unique( this.currentTaxonomies[ taxonomy ] ).join( ', ' );
+		}
+
+		return '';
+	};
+
+	/**
+	 * replaces the category strings with the category names.
+	 *
+	 * @param {String} data
+	 * @return {String}
+	 */
+	YoastReplaceVarPlugin.prototype.tagReplace = function( data ) {
+		if ( jQuery( '#post_ID').val() !== undefined ) {
+			data = data.replace( /%%tag%%/g, jQuery( ' #tax-input-post_tag' ).val() );
+		}
+
+		return data;
+	};
+
+	/**
+	 * Replaces the taxonomy description
+	 *
+	 * @param {String} data
+	 * @return {String}
+	 */
+	YoastReplaceVarPlugin.prototype.taxonomyDescriptionReplace = function( data ) {
+		var text = YoastSEO.app.rawData.text;
+
+		data = data.replace( /%%category_description%%/g, text );
+		data = data.replace( /%%tag_description%%/g, text );
+		data = data.replace( /%%term_description%%/g, text );
+
+		// We want to replace 'ct_desc_{custom_taxonomy_name}' with the value of `text`.
+		if ( wpseoReplaceVarsL10n.taxonomy_description !== '' ) {
+			data = data.replace( '%%' + wpseoReplaceVarsL10n.taxonomy_description + '%%', text );
+		}
+
+		return data;
+	};
+
+	/**
+	 * Fetch the taxonomies, based on the checked checkboxes
+	 *
+	 * @param {object} targetMetaBox
+	 *
+	 * @return {void}
+	 */
+	YoastReplaceVarPlugin.prototype.fetchTaxonomies = function( targetMetaBox ) {
+		var activeCheckboxes = jQuery( targetMetaBox ).find( 'input:checked' );
+		var taxonomyName     = jQuery( targetMetaBox ).attr( 'id' ).replace( 'taxonomy-', '' );
+
+		if ( activeCheckboxes.length > 0 ) {
+			this.setUnparsedTaxonomies( activeCheckboxes, taxonomyName );
+		}
+
+		this.setCurrentTaxonomies( activeCheckboxes, taxonomyName );
+		this.declareReloaded();
+	};
+
+	/**
+	 * Filters the already fetch categories
+	 *
+	 * @param {Object[]} activeCheckboxes
+	 */
+	YoastReplaceVarPlugin.prototype.setUnparsedTaxonomies = function( activeCheckboxes, taxonomyName ) {
+		if ( this.taxonomyRepository[ taxonomyName ] === undefined ) {
+			this.taxonomyRepository[ taxonomyName ] = [];
+		}
+
+		activeCheckboxes.each(
+			function( index, checkbox ) {
+				var taxonomyID = jQuery(checkbox).val();
+				if ( this.taxonomyRepository[ taxonomyName ][ taxonomyID ] === undefined ) {
+					this.taxonomyRepository[ taxonomyName ][ taxonomyID ] = this.getTaxonomyName( checkbox );
+				}
+			}.bind( this )
+		);
+	};
+
+	/**
+	 * Gets the taxonomy name.
+	 *
+	 * @param {Object} checkbox
+	 * @returns String
+	 */
+	YoastReplaceVarPlugin.prototype.getTaxonomyName = function( checkbox ) {
+		return jQuery(checkbox).parent( 'label' ).clone().find('span').remove().end().text().trim();
```

Comment:
This should be abstracted to a function with a good docblock. This line is very unclear and very hard to parse.

With the link of the source of the fix: http://viralpatel.net/blogs/jquery-get-text-element-without-child-element/